executor/infoschema_reader.go

```diff
@@ -78,6 +78,9 @@ type memtableRetriever struct {
 
 // retrieve implements the infoschemaRetriever interface
 func (e *memtableRetriever) retrieve(ctx context.Context, sctx sessionctx.Context) ([][]types.Datum, error) {
+	if e.table.Name.O == infoschema.TableClusterInfo && !hasPriv(sctx, mysql.ProcessPriv) {
+		return nil, plannercore.ErrSpecificAccessDenied.GenWithStackByArgs("Process")
```

Comment:
This breaks the telemetry unit test :(
Need to check how to handle the telemetry code to have the correct privileges.

Reply:
That error in `testSuite.Test02Preview` comes from the ExecRestrictedStmt binding with no privilegeManager instance.

func hasPriv(ctx sessionctx.Context, priv mysql.PrivilegeType) bool {
	if pm := privilege.GetPrivilegeManager(ctx); pm != nil {
		return pm.RequestVerification(ctx.GetSessionVars().ActiveRoles, "", "", "", priv)
	}
	return false
}

that's why it failed here, and return nothing.

Reply:
I've talked with @djshow832 , @breeswish and other engineers, we prefer to give the internal SQL executor(`ExecRestrictedStmt`) higher privilege to access those tables.

Is there an easy way to archive it?

Reply:
Looks that we need to grant the related privileges(`CONFIG`/`PROCESS`) to:
- Sessions used by the test cases:  https://github.com/pingcap/tidb/blob/0f151af7c5f86e205ff726ee707b7a2e9bb8844f/telemetry/telemetry_test.go#L63.
- Session used in the real case: https://github.com/pingcap/tidb/blob/118671d990c7641503938cb278b53bf7442e9a34/session/session.go#L2559.

Reply:
I am not completely sure we should give the telemetry the `Process` privilege, since it will then have access to running queries which may leak sensitive data. But on the other hand, it is still running within the server so it could as well have full access.
I did not find a quick way/small patch to fix it, so I will dig deeper later unless someone is helpful and can provide a patch :)

Reply:
I found the problems comes from 
```
// internal session created with createSession doesn't has the PrivilegeManager. For most experienced cases before,
// we use it like this:
// ```
// checker := privilege.GetPrivilegeManager(ctx)
// if checker != nil && !checker.RequestVerification(ctx.GetSessionVars().ActiveRoles, schema.Name.L, table.Name.L, "", mysql.AllPrivMask) {
//	  continue
// }
// do something.
// ```
// So once the privilege manager is nil, it's a signature of internal sql, so just passing the checker through.
```